### ISSUE TITLE
Server working fine until abruptly slowing down

### ISSUE THREAD

**Issue by lukaljoncol** [DarkPacks/SevTech-Ages#4658]
## Bug Description

 Server works for a little while until abruptly and quickly deteriorating from 20tps to ~0 over the course of about 5 minutes.

### Did this behavior use to work in the previous version?

 Yes, any previous servers used to run fine.

## Steps to Reproduce (for bugs)

1. Start server
2. Wait

## Logs

* Client/Server Log: https://pastebin.com/iRguMF19 (wasn't able to upload entire log, so this is everything that happens after the server is opened and joinable)
* Crash Log: No crash logs are created, server ceases to respond to any commands.

## Client Information
* Modpack Version: 3.2.1
* Java Version: Java 8 Update 202 (64-bit)
* Launcher Used: MultiMC
* Memory Allocated: 10GB

## World Information
* Modpack Version world created in: 3.2.1

## Server Information
* Java Version: Java 8 Update 202 (64-bit)
* Operating System: Windows 10
* Hoster/Hosting Solution: Self hosting with 12 GB allocated
* Sponge (Non-Vanilla Forge) Server: No

I should note that initially we had thought the problem to be a chunk error, as previous Sevtech worlds we have hosted with server crashes have been solved by rigorously searching for and deleting chunks that were problematic. However, I suspect this time the problem might be caused by a memory leak (keep in mind I'm not extremely experienced with Java so I could be wrong). I've tried allocating more RAM to the server (I think up to 16GB) to no avail. I'm familiar with using Timings reports on Spigot, but I'm not sure how I would be able to get something similar to that or a heap dump on a forge server. I did take two debug profiler reports, one while the server was running fine and another when the server was running very slow.

Healthy server: https://pastebin.com/RTayRTmB
Unhealthy server: https://pastebin.com/mwDjP43x

**Comment by sam-kirby** [comment]
Do you observe the same issue on SevTech 3.2.3?

There is a memory leak in 3.2.1 whereby item entities are not cleaned up from memory even after they've despawned or been picked up. This usually affects long running servers, but if you have some fast farm or similar on the server this may lead to a crash much sooner. This was resolved in 3.2.3.

There are numerous profiling options available on Forge; LagGoggles, Spark and Sampler (from least to most powerful, or most to least user friendly).

**Comment by lukaljoncol** [comment]
I'm trying 3.2.3 right now, I had suspected some sort of memory problem and you mentioning that there was a problem with long running servers on 3.2.1 is very affirming.

**Comment by lukaljoncol** [comment]
Unfortunately updating to 3.2.3 didn't change much. I'm attempting to use Spark now.

**Comment by lukaljoncol** [comment]
Yeah... can't seem to find anything specific. I'm only around and able to test for the next few days so it's unlikely I'm going to be able to fix this, but the server's been unplayable since June of this year. I managed to get some Spark profiles but they're out of my understanding in terms of finding a solution. If anyone wants to take a stab at them I've provided a .zip with profiles taken during a period of no lag and 15 minutes later when it was lagging.

[sparkprofiles.zip](https://github.com/DarkPacks/SevTech-Ages/files/7791231/sparkprofiles.zip)

**Comment by sam-kirby** [comment]
I'm not able to draw any conclusions from this either unfortunately, a heap summary might be more helpful though; see https://spark.lucko.me/docs/Command-Usage#spark-heapsummary

**Comment by lukaljoncol** [comment]
Idk if you're still around, I got a heap summary though
[heapsummary.txt](https://github.com/DarkPacks/SevTech-Ages/files/7937268/heapsummary.txt)

**Comment by lukaljoncol** [comment]
A crash report too....
[crash-2022-02-01_04.34.53-server.txt](https://github.com/DarkPacks/SevTech-Ages/files/7977346/crash-2022-02-01_04.34.53-server.txt)

**Comment by sam-kirby** [comment]
You appear to be on version 3.2.1 - this issue should have been resolved in 3.2.3.
